src/actions/general.ts

```diff
@@ -200,6 +200,37 @@ export function getWarnMetricsStatus(): ActionFunc {
     };
 }
 
+export function setFirstAdminVisitMarketplaceStatus(): ActionFunc {
+    return async (dispatch: DispatchFunc) => {
+        try {
+            Client4.trackEvent('api', 'api_request_set_first_admin_visit_marketplace_status');
```

Comment:
I'm 3/5 that we should track event on the components instead of Redux. That simplifies the number of place to look for a track event and seems to be the idiomatic way in the code base.

Reply:
I'm not particular to one or another (an action is an action), but I'd like to know more about: "seems to be the idiomatic way" - I don't see a definitive pattern one way or another - can you help substantiate the statement?

Reply:
My initiation comment was a misleading. IMO the event category should be `plugins` because that is what all other marketplace events are also using e.g. `ui_marketplace_search` or `ui_marketplace_opened`. Therefore the event should triggered on the component, as only `api` events are triggered in Redux.

Reply:
Ok, I get your comment now. I can see the benefit of using the event category 'plugins', wrt to the redux vs. component, don't really mind one vs. another since there is a 1-1 mapping of the UI action to the redux action (and apparently redux captures other categories besides `api`), but one could argue that the first/original action is the UI one so I'll move it there.